<launch>
    <!-- deliver_to and pickup_package action server -->
    <node
        pkg="jsk_spot_delivery_demo"
        type="delivery_action_server.py"
        name="delivery_action_server"
        output="screen">
        <rosparam>
            force_threshold: 5.0
        </rosparam>
        <rosparam command="load" file="$(find spot_behavior_graph)/config/nodes.yaml" />
        <remap from="~person_visible" to="/delivery_demo_person_tracker/visible" />
        <remap from="~people_pose_array" to="/delivery_demo_person_tracker/people_pose_array" />
        <remap from="~reset_body_force" to="/body_force_publisher/zero_offset" />
        <remap from="~body_force" to="/body_force_publisher/wrench_output" />
    </node>

    <!-- body force caculation -->
    <node
        pkg="jsk_spot_delivery_demo"
        type="body_force_publisher.py"
        name="body_force_publisher"
        output="screen">
        <remap from="~input_wrench_front_right" to="/front_right/wrench" />
        <remap from="~input_wrench_front_left" to="/front_left/wrench" />
        <remap from="~input_wrench_rear_right" to="/rear_right/wrench" />
        <remap from="~input_wrench_rear_left" to="/rear_left/wrench" />
    </node>

    <group ns="front_right">
        <node
            pkg="virtual_force_publisher"
            type="virtual_force_publisher"
            name="virtual_force_publisher">
            <remap from="joint_states" to="/joint_states" />
            <rosparam subst_value="true">
                publish_frequency: 50.0
                time_constant: 0.3
                root: body
                tip: front_right_lower_leg
            </rosparam>
        </node>
    </group>

    <group ns="front_left">
        <node
            pkg="virtual_force_publisher"
            type="virtual_force_publisher"
            name="virtual_force_publisher">
            <remap from="joint_states" to="/joint_states" />
            <rosparam subst_value="true">
                publish_frequency: 50.0
                time_constant: 0.3
                root: body
                tip: front_left_lower_leg
            </rosparam>
        </node>
    </group>

    <group ns="rear_right">
        <node
            pkg="virtual_force_publisher"
            type="virtual_force_publisher"
            name="virtual_force_publisher">
            <remap from="joint_states" to="/joint_states" />
            <rosparam subst_value="true">
                publish_frequency: 50.0
                time_constant: 0.3
                root: body
                tip: rear_right_lower_leg
            </rosparam>
        </node>
    </group>

    <group ns="rear_left">
        <node
            pkg="virtual_force_publisher"
            type="virtual_force_publisher"
            name="virtual_force_publisher">
            <remap from="joint_states" to="/joint_states" />
            <rosparam subst_value="true">
                publish_frequency: 50.0
                time_constant: 0.3
                root: body
                tip: rear_left_lower_leg
            </rosparam>
        </node>
    </group>
</launch>
